### PR TITLE
feat: add leaderboard view and component

### PR DESCRIPTION
**Pull request by rellen**
- add a leaderboard component
- add a leaderboard calculation to Game
- add a slug field to Game
- add a leaderboard view navigable via Game's slug

### CODE REVIEW

File: lib/playdate_web/live/live_helpers.ex [hunk 1]
```diff
@@ -0,0 +1,17 @@
+defmodule PlaydateWeb.Live.LiveHelpers do
+  @moduledoc """
+  Custom LiveView helpers
+  """
+
+  @doc """
+  Tag a socket with an :ok tuple (useful for the mount callback)
+  """
+  @spec ok(Phoenix.Socket.t()) :: {:ok, Phoenix.Socket}
```
Comment (juddey) [inline lib/playdate_web/live/live_helpers.ex:9]: Ha! I was going to ask should we introduce these!

File: mix.exs [hunk 2]
```diff
@@ -66,6 +66,8 @@ defmodule Playdate.MixProject do
       {:guardian, "~> 2.2"},
       {:bcrypt_elixir, "~> 3.0"},
       {:petal_components, "~> 0.17"},
+      {:mix_test_interactive, "~> 1.2"},
```
Comment (juddey) [inline mix.exs:69]: And this one too :) \o/

**Review by juddey — APPROVED** [review]
LGTM, I'll leave it to @jimsynz to say if we're going to go with the factory pattern for tests, I have some vague recollection we discussed it.

File: lib/playdate_web/live/leaderboard_live.ex [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  use PlaydateWeb, :live_view
+
+  alias PlaydateWeb.Components.Leaderboard
+
+  alias Playdate.Games.Game
+
+  @impl true
+  def mount(_params, session, socket) do
+    socket |> assign(:current_user, session["current_user"]) |> assign(:leaders, []) |> ok()
+  end
```
Comment (jimsynz) [inline lib/playdate_web/live/leaderboard_live.ex:11]: Oh. It needs the current user for the header, right.  Maybe use a `live_session` in the router instead? there's already a `:live_user` option in the `LiveUserAuth` module, so you can do:

```
# router.ex
live_session :maybe_authenticated,
      on_mount: {PlaydateWeb.LiveUserAuth, :live_user} do
      live ...
end

**Pull request merged**